optaplanner-core/src/test/java/org/optaplanner/core/impl/score/director/DummyConstraintProvider.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.director;
+
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.core.impl.testdata.domain.TestdataEntity;
+
+public final class DummyConstraintProvider implements ConstraintProvider {
+    @Override
+    public Constraint[] defineConstraints(ConstraintFactory constraintFactory) {
+        return new Constraint[] {
+                onlyConstraint(constraintFactory)
+        };
+    }
+
+    private Constraint onlyConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(TestdataEntity.class)
+                .rewardConfigurable("First weight");
```

Comment:
Ow wait, this isn't a dummy, it actually does something.
 This is a TestdataSolutionConstraintProvider as it only works for TestdataSolutio

Reply:
And it does something for a @ConstraintWeight specifically, so it's naming so mention that.

how about TestdataConstraintWeightConstraintProvider ? Or something better :)